service/src/main/java/com/ge/predix/acs/policy/evaluation/cache/AbstractPolicyEvaluationCache.java

```diff
@@ -67,6 +96,14 @@ public PolicyEvaluationResult get(final PolicyEvaluationRequestCacheKey key) {
         } catch (IOException e) {
             throw new IllegalStateException("Failed to read policy evaluation result as JSON.", e);
         }
+        // Remove the Policy Evaluation Result out of values because we deserialized it
+        values.remove(values.size() - 1);
+
+        String zone = key.getZoneId();
+        // Add the resolved resource URI's stored in the cachedResult to values so that we can check timestamps of
+        // all entities involved in the request.
+        values.addAll(multiGet(cachedResult.getResolvedResourceUris().stream()
+                .map(resourceUri -> resourceKey(zone, resourceUri)).collect(Collectors.toList())));
 
```

Comment:
One optimization we could do here is to compare resource from request with resolved resource.  If policy doesn't contain any template URIs then request resource and resolved resource are the same.  In that case we don't have to perform a multiGet again.  In that case when assembling keys for the first call to multiGet we need to include a resource from request again.